From b41a2a1ef9d64caf83716092162f4afe856a5a7a Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Tue, 25 Oct 2022 14:31:01 +0800
Subject: [PATCH 2/7] iio: adc: npcm: fix inappropriate error log

npcm_adc_read may return -ERESTARTSYS (interrupted by a signal),
don't print error log since this is not an error condition.

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 drivers/iio/adc/npcm_adc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/adc/npcm_adc.c b/drivers/iio/adc/npcm_adc.c
index d41f8cab7ce9..743050a70368 100644
--- a/drivers/iio/adc/npcm_adc.c
+++ b/drivers/iio/adc/npcm_adc.c
@@ -347,7 +347,8 @@ static int npcm_adc_read_raw(struct iio_dev *indio_dev,
 		ret = npcm_adc_read(info, val, chan->channel);
 		mutex_unlock(&info->lock);
 		if (ret) {
-			dev_err(info->dev, "NPCM ADC read failed\n");
+			if (ret != -ERESTARTSYS)
+				dev_err(info->dev, "NPCM ADC read failed\n");
 			return ret;
 		}
 
-- 
2.43.0

